dev/docs/supported-features.md

```diff
@@ -2,8 +2,8 @@
 Below is a list of the Kubernetes Gateway API features supported in the current release of the
 Consul API Gateway.
 
-Consul API Gateway version: **v0.2.0**
-Suppoorted K8s Gateway API version: **v1alpha2**
+Consul API Gateway version: **v0.4.0**
```

Comment:
This will get bumped as part of our pre-release automation once #310 merges so that it shouldn't go out of date in the future